### PR TITLE
[local gc]  move DacNotifyGCMarkEnd and AnalyzeSurvivorsRequested to GCToEEInterface 

### PR DESCRIPTION
**Pull request by davmason**
Fix for #17771

### CODE REVIEW

File: src/gc/env/gcenv.ee.h [hunk 1]
```diff
@@ -92,6 +92,9 @@ class GCToEEInterface
     static uint32_t GetIndexOfAppDomainBeingUnloaded();
     static uint32_t GetTotalNumSizedRefHandles();
     static bool AppDomainIsRudeUnload(void *appDomain);
+
+    static bool AnalyzeSurvivorsRequested(int condemnedGeneration);
+    static void DACNotifyGcMarkEnd(int condemnedGeneration);
```
Comment (jkotas) [inline src/gc/env/gcenv.ee.h]: Should this rather be called AnalyzeSurvivorsFinished?

**Comment by davmason** [comment]
@mmitche I am only seeing 10 CI jobs here, when there is usually around 30.  I'm not sure how to debug what is preventing the rest from appearing, is there something affecting the CI?

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@davmason Something is up with the main CI instance.  high CPU usage is preventing coreclr from starting all its jobs.  I'm looking into it.

**Comment by davmason** [comment]
@mmitche Thanks!

**Comment by davmason** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot test Windows_NT x64 Release CoreFX Tests

**Pull request merged**